### ISSUE TITLE
THEPeerBluetooth.m needs cleanup

### ISSUE THREAD

**Issue by yaronyg** [thaliproject/Thali_CordovaPlugin#96]
This is an enormous mess of code. It's hard to follow because of the number of things it is doing. It desperately needs refactoring.

**Comment by tobybrad** [comment]
I'd love to rewrite the whole stack for 3.0 in Swift with a view to releasing as in independent lib.

**Comment by yaronyg** [comment]
I want a pony.

**Comment by yaronyg** [comment]
Seriously though I could see this work as part of story 8 (http://thaliproject.org/stories) but that is pretty far off.

**Comment by yaronyg** [comment]
This code is going away with the swift rewrite
